pkg/clients/roadmap_client/roadmap.go

```diff
@@ -107,3 +120,89 @@ func (rc roadmapClient) GetAppstreams(ctx context.Context) (AppstreamsResponse,
 
 	return appStreamResp, statusCode, nil
 }
+
+func (rc roadmapClient) GetRhelLifecycle(ctx context.Context) (LifecycleResponse, int, error) {
+	statusCode := http.StatusInternalServerError
+	server := config.Get().Clients.Roadmap.Server
+	var err error
+	var lifecycleResponse LifecycleResponse
+	var body []byte
+
+	appstreams, err := rc.cache.GetRoadmapRhelLifecycle(ctx)
+	if err != nil && !errors.Is(err, cache.NotFound) {
+		log.Error().Err(err).Msg("GetAppstreams - error reading from cache")
+	}
+	if appstreams != nil {
+		err = json.Unmarshal(appstreams, &lifecycleResponse)
+		if err != nil {
+			return LifecycleResponse{}, statusCode, fmt.Errorf("error during unmarshal response body: %w", err)
+		}
+		return lifecycleResponse, http.StatusOK, nil
+	}
+
+	fullPath := server + "/lifecycle/rhel"
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, fullPath, nil)
+	if err != nil {
+		return LifecycleResponse{}, statusCode, fmt.Errorf("error building request: %w", err)
+	}
+
+	if config.Get().Clients.Roadmap.Username != "" && config.Get().Clients.Roadmap.Password != "" {
+		req.SetBasicAuth(config.Get().Clients.Roadmap.Username, config.Get().Clients.Roadmap.Password)
+	}
+
+	encodedXRHID, err := encodedIdentity(ctx)
+	if err != nil {
+		return LifecycleResponse{}, statusCode, fmt.Errorf("error getting encoded XRHID: %w", err)
+	}
+	req.Header.Set(api.IdentityHeader, encodedXRHID)
+
+	resp, err := rc.client.Do(req)
+	if resp != nil {
+		defer resp.Body.Close()
+
+		body, err = io.ReadAll(resp.Body)
+		if err != nil {
+			return LifecycleResponse{}, statusCode, fmt.Errorf("error reading response body: %w", err)
+		}
+		if resp.StatusCode != 0 {
+			statusCode = resp.StatusCode
+		}
+	}
+	if err != nil {
+		return LifecycleResponse{}, statusCode, fmt.Errorf("error sending request: %w", err)
+	}
+	if statusCode < 200 || statusCode >= 300 {
+		return LifecycleResponse{}, statusCode, fmt.Errorf("unexpected status code with body: %s", string(body))
+	}
+
+	rc.cache.SetRoadmapRhelLifecycle(ctx, body)
+
+	err = json.Unmarshal(body, &lifecycleResponse)
+	if err != nil {
+		return LifecycleResponse{}, statusCode, fmt.Errorf("error during unmarshal response body: %w", err)
+	}
+
+	return lifecycleResponse, statusCode, nil
+}
+
+func (rc roadmapClient) GetRhelLifecycleForLatestMajorVersions(ctx context.Context) (map[int]LifecycleEntity, error) {
+	lifecycleResp, _, err := rc.GetRhelLifecycle(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	currentDate := time.Now().UTC().Truncate(24 * time.Hour)
+	rhelEolMap := make(map[int]LifecycleEntity)
+	for _, item := range lifecycleResp.Data {
+		startDate, err := time.Parse("2006-01-02", item.StartDate)
+		if err != nil {
+			return nil, err
+		}
+		if startDate.Before(currentDate) || startDate.Equal(currentDate) {
+			if existing, found := rhelEolMap[item.Major]; !found || (item.Minor > existing.Minor) {
+				rhelEolMap[item.Major] = item
+			}
```

Comment:
chatted privately, but we actually want to pick the earliest start date and latest end date for a given RHEL major release (9)